### ISSUE TITLE
Remarks

### ISSUE THREAD

**Issue by jacob-wg2** [sebastiw/esock#1]
Looks great.

Some minor issues (to be continued):

https://github.com/sebastiw/sock/blob/cbcb23ab699a51c6bfef2798c40ee0ec0d6566dc/src/sock.erl#L62-L63

Binding to all local interfaces by default is probably problematic, especially when local IP addresses are used.  Which local IP addresses are announced to the peer in that case? What happens if a new interface is added or an existing goes down/up?

https://github.com/sebastiw/sock/blob/cbcb23ab699a51c6bfef2798c40ee0ec0d6566dc/src/sock_assoc.erl#L62-L64

Wondering what dialyzer will say depending on USE_SOCKET. Why not using `-ifdef`?

https://github.com/sebastiw/sock/blob/cbcb23ab699a51c6bfef2798c40ee0ec0d6566dc/src/sock_utils.erl#L26-L27

Wouldn't cope with inet6 and shortened addresses (less than 3 dots).

What about having a function that takes two inet:ip_address and a number (mask) instead? Or at least using net:parse_address internally instead of split?

**Comment by sebastiw** [comment]
1. for testing it's excellent :) I don't hope anyone will use that in prod. all `create_ep/0-3` are exported so one can choose to use any.

2. Updated with `-ifdef`

3. Fixed for ipv6 and shortened addresses, though no function actually use it yet.

**Comment by sebastiw** [comment]
I did update 1. to loopback instead

**Comment by lixen-wg2** [comment]
I don't think we should use the supervisor as a lookup. We have seen that it doesn't scale. I also wonder what happens when you have failures and which_children messages filling the message box of the supervisor
https://github.com/sebastiw/sock/blob/16c022414a677310a13340b1d2a2e43ac41ac24e/src/sock_sup.erl#L32



Should we do recv or {active, N} ? I think we greatly increased performance in diameter testing by having active tuned.
https://github.com/sebastiw/sock/blob/16c022414a677310a13340b1d2a2e43ac41ac24e/src/sock_ep.erl#L154



Does it make sense to move this read outside gen_server?
https://github.com/sebastiw/sock/blob/main/src/sock_reg.erl#L84
